File: javascript/fibonacci.js
function fibonacci(num) {
  let first_two = [0, 1]

  for (let i = (num - 1); i > 0; i--) {
    const sum = first_two[0] + first_two[1]

    first_two = [first_two[1], sum]
  }

  return first_two[1]
}

if (require.main === module) {
  // add your own tests in here
  console.log("Expecting: 0");
  console.log("=>", fibonacci(0));

  console.log("");

  console.log("Expecting: 1");
  console.log("=>", fibonacci(2));

  console.log("");

  console.log("Expecting: 55");
  console.log("=>", fibonacci(10));
}

module.exports = fibonacci;

// Please add your pseudocode to this file
// And a written explanation of your solution

File: ruby/fibonacci.rb
def fibonacci(num)
  first_two = [0, 1]

  (num - 1).times do 
    sum = first_two[0] + first_two[1]
    first_two = [first_two[1], sum]
  end

  first_two.last
end

if __FILE__ == $PROGRAM_NAME
  puts "Expecting: 0"
  puts "=>", fibonacci(0)

  puts

  puts "Expecting: 1"
  puts "=>", fibonacci(2)

  puts

  puts "Expecting: 55"
  puts "=>", fibonacci(10)

  # Don't forget to add your own!
end

# Please add your pseudocode to this file
# And a written explanation of your solution
